elf2table/src/lib.rs

```diff
@@ -75,7 +86,9 @@ pub fn parse(elf: &[u8]) -> Result<Option<Table>, anyhow::Error> {
         }
     };
 
-    defmt_decoder::check_version(version).map_err(anyhow::Error::msg)?;
+    if check_version {
```

Comment:
personally, I'd rather pull this out of `parse_impl()` and move it into `parse()` right before the call to `parse_impl()`. That way we can avoid having a flag argument* as well as tight coupling between the parsing and defmt consistency check, which imo are two different tasks.

* the function names of `parse_ignore_version()` and `parse()` imo already communicate what's happening better than parse(.. , true) does

Reply:
Doing the version check requires some amount of processing to get the `version`, so I'd have to duplicate that into the caller.

> tight coupling between the parsing and defmt consistency check, which imo are two different tasks.

There is some amount of intentional coupling between them, for example, we always check that there's only one version in use, and we check for the existence of the `.defmt` section and the version symbol to determine whether defmt is in use at all, or the linker script is broken.

This PR is only intended to skip the version check, not any of the other consistency checks.

Reply:
hrm , ok then that's maybe one for future refactoring sessions...